finagle-core/src/main/scala/com/twitter/finagle/tracing/TraceInitializerFilter.scala

```diff
@@ -243,11 +243,24 @@ private[finagle] object WireTracingFilter {
 
   def module[Req, Rep]: Stackable[ServiceFactory[Req, Rep]] =
     new Stack.Module1[param.Label, ServiceFactory[Req, Rep]] {
-      val role = ClientTracingFilter.role
+      val role = WireTracingFilter.role
       val description = "Report finagle information and wire send/recv events"
       def make(_label: param.Label, next: ServiceFactory[Req, Rep]) = {
         val param.Label(label) = _label
         TracingFilter[Req, Rep](label) andThen next
       }
     }
+
+  def serverModule[Req, Rep]: Stackable[ServiceFactory[Req, Rep]] =
+    new Stack.Module0[ServiceFactory[Req, Rep]] {
+      val role = WireTracingFilter.role
+      val description = "Report wire recv/send events"
+      def make(next: ServiceFactory[Req, Rep]) = filter andThen next
+      val filter = Filter.mk[Req, Rep, Req, Rep] { (req, svc) =>
```

Comment:
Bleh you're right I was getting confused with WireRecvError which does not
exist.  Still unsure if this makes sense semantically (it wouldn't be an
error encountered while trying to send on the wire)

On Friday, July 24, 2015, Antoine Tollenaere <notifications@github.com>
wrote:

> In
> finagle-core/src/main/scala/com/twitter/finagle/tracing/TraceInitializerFilter.scala
> <https://github.com/twitter/finagle/pull/401#discussion_r35445892>:
>
> >        val description = "Report finagle information and wire send/recv events"
> >        def make(_label: param.Label, next: ServiceFactory[Req, Rep]) = {
> >          val param.Label(label) = _label
> >          TracingFilter[Req, Rep](label) andThen next
> >        }
> >      }
> > +
> > +  def serverModule[Req, Rep]: Stackable[ServiceFactory[Req, Rep]] =
> > +    new Stack.Module0[ServiceFactory[Req, Rep]] {
> > +      val role = WireTracingFilter.role
> > +      val description = "Report wire recv/send events"
> > +      def make(next: ServiceFactory[Req, Rep]) = filter andThen next
> > +      val filter = Filter.mk[Req, Rep, Req, Rep] { (req, svc) =>
>
> Why not annotate on failures (with a new WireSendError(error: String))
> here and reuse AnnotatingTracingFilter?
>
> Otherwise LGTM.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twitter/finagle/pull/401/files#r35445892>.
>


-- 
Oliver Gould <ver@buoyant.io>